### PR TITLE
Zoidberg improvements

### PR DESCRIPTION
**Pull request by dschwoerer**

### CODE REVIEW

File: tools/pylib/zoidberg/field.py [hunk 1]
```diff
@@ -1595,7 +1615,16 @@ def field_values(r, phi, z, configuration=0, plot_poincare=False):
                 pos.x3 = x3[slc]
 
                 ## Call tracer service
-                res = tracer.service.magneticField(pos, config)
+                redo = 2
+                while redo:
+                    try:
+                        res = tracer.service.magneticField(pos, config)
+                    except:
```
Comment (ZedThree) [inline tools/pylib/zoidberg/field.py:1622]: Can you be explicit in what exception it catches?

File: tools/pylib/zoidberg/field.py [hunk 2]
```diff
@@ -1595,7 +1615,16 @@ def field_values(r, phi, z, configuration=0, plot_poincare=False):
...
+                        redo -= 1
```
Comment (ZedThree) [inline tools/pylib/zoidberg/field.py:1623]: Putting a short `sleep` here might be wise, if this is to help with intermittent network failures

File: tools/pylib/zoidberg/field.py [hunk 3]
```diff
@@ -1549,8 +1555,22 @@ def field_values(r, phi, z, configuration=0, plot_poincare=False):
...
+        elif os.path.isfile(fname_old):
+            print("Saved field found, loading from: ", fname_old)
+            with open(fname_old, pickle_read_mode) as f:
                 Br, Bphi, Bz = pickle.load(f)
+            ds = xr.Dataset()
+            ds["Br"] = ("x", "y", "z"), Br
+            ds["Bz"] = ("x", "y", "z"), Bz
+            ds["Bphi"] = ("x", "y", "z"), Bphi
+            ds.to_netcdf(fname)
+            del ds
```
Comment (ZedThree) [inline tools/pylib/zoidberg/field.py:1573]:
```suggestion
            with xr.Dataset() as ds:
                ds["Br"] = ("x", "y", "z"), Br
                ds["Bz"] = ("x", "y", "z"), Bz
                ds["Bphi"] = ("x", "y", "z"), Bphi
                ds.to_netcdf(fname)
```

File: tools/pylib/zoidberg/field.py [hunk 4]
```diff
@@ -1612,9 +1641,15 @@ def field_values(r, phi, z, configuration=0, plot_poincare=False):
...
+            del By
+
             ## Save so we don't have to do this every time.
-            with open(fname, pickle_write_mode) as f:
-                pickle.dump([Br, Bphi, Bz], f)
+            ds = xr.Dataset()
+            ds["Br"] = ("x", "y", "z"), Br
+            ds["Bz"] = ("x", "y", "z"), Bz
+            ds["Bphi"] = ("x", "y", "z"), Bphi
+            ds.to_netcdf(fname)
```
Comment (ZedThree) [inline tools/pylib/zoidberg/field.py:1652]:
```suggestion
            with xr.Dataset() as ds:
                 ds["Br"] = ("x", "y", "z"), Br
                 ds["Bz"] = ("x", "y", "z"), Bz
                 ds["Bphi"] = ("x", "y", "z"), Bphi
                 ds.to_netcdf(fname)
```

**Review by ZedThree — APPROVED** [review]
LGTM, a few places could be polished a little.

It looks like `black` has been applied to the whole codebase, which makes the diff a little painful, so looking at the individual commits is easier.

**Comment by dschwoerer** [comment]
I rebased to get rid of the black-ci commit, that caused merge conflicts.

**Comment by dschwoerer** [comment]
Replaced by:

https://github.com/boutproject/zoidberg/pull/3

**Pull request closed without merging**